File: Java/src/main/java/edu/cmu/cs/cs214/rec04/InheritanceSortedIntList.java
package edu.cmu.cs.cs214.rec04;

import java.nio.channels.InterruptibleChannel;
import java.util.ArrayList;
import java.util.List;

/**
 * InheritanceSortedIntList -- a variant of a SortedIntList that keeps
 * count of the number of attempted element insertions (not to be confused
 * with the current size, which goes down when an element is removed)
 * and exports an accessor (totalAdded) for this count.
 *
 * @author Nora Shoemaker
 *
 */

public class InheritanceSortedIntList extends AbstractIntList {
    private int count;
    private List<Integer> data;

    public InheritanceSortedIntList(){
        this.data = new ArrayList<Integer>();
        this.count = 0;
    }

    @Override
    public boolean add(int num){
        count++;
        return data.add(num);
    }

    public int get(int index){
        return data.get(index);
    }

    public boolean remove(int i){
        return data.remove((Integer)i);
    }


    public int size(){
        return data.size();
    }
    

    public int getTotalAdded(){
        return count;
    }

}

File: Typescript/src/InheritanceSortedIntList.ts
import { AbstractIntList } from './AbstractIntList.js'
import { SortedIntList } from './hidden/SortedIntListLibrary.js'

/**
 * InheritanceSortedIntList -- a variant of a SortedIntList that keeps
 * count of the number of attempted element insertions (not to be confused
 * with the current size, which goes down when an element is removed)
 * and exports an accessor (totalAdded) for this count.
 *
 * @author Nora Shoemaker
 *
 */

class InheritanceSortedIntList extends SortedIntList{
  private count : number = 0;
  
  constructor(){
    super();
    this.count = 0;
  }

  public add (num: number): boolean{
    this.count++;
    return super.add(num);
  }
  
  public getTotalAdded() : number{
    return this.count;
  }

}

export { InheritanceSortedIntList }
